libs/payments/ui/src/lib/client/components/SubscriptionContent/index.tsx

```diff
@@ -64,8 +93,144 @@ export const SubscriptionContent = ({
     false,
     locale
   );
+
+  async function resubscribe() {
+    setPendingResubscribe(true);
+    setResubscribeActionError(false);
+
+    const result = await resubscribeSubscriptionAction(userId, subscription.id);
+    if (result.ok) {
+      setOpenResubscribeDialog(false);
+      // TODO: This is a workaround to match existing legacy behavior.
+      // Fix as part of redesign
+      await new Promise((resolve) => setTimeout(resolve, 500));
+      setOpenResubscribeSuccessDialog(true);
+    } else {
+      setResubscribeActionError(true);
+    }
+    setPendingResubscribe(false);
+  }
+
   return (
     <>
+      <Dialog.Root open={openResubscribeSuccessDialog}>
+        <Dialog.Portal>
+          <Dialog.Overlay className='fixed inset-0 bg-black/75 z-50' />
+          <Dialog.Content
+            className='w-11/12 max-w-[545px] text-center px-7 pt-6 pb-8 rounded-xl shadow inline-block fixed top-1/2 left-1/2 -translate-x-1/2 -translate-y-1/2 z-9999 bg-white'
+            onEscapeKeyDown={() => setOpenResubscribeSuccessDialog(false)}
+            onPointerDownOutside={() => setOpenResubscribeSuccessDialog(false)}
+            onInteractOutside={() => setOpenResubscribeSuccessDialog(false)}
+          >
+            <Dialog.Title className='font-bold leading-6 m-5 space-y-5'>
+              <Image src={webIcon} alt={productName} height={64} width={64} className='h-16 w-16 mx-auto' />
+              <Localized
+                id="resubscribe-success-dialog-title"
+              >
+                <p>Thanks! You’re all set.</p>
+              </Localized>
+            </Dialog.Title>
+            <Dialog.Description className='leading-6 space-y-4'>
```

Comment:
Suggestion: improvements for accessibility
`id="resubscribe-success-description"` to `Dialog.Description`